### PR TITLE
chore: make logout observable (AR-1924)

### PR DESCRIPTION
**Pull request by vitorhugods**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

One of the criterias to stop sync automatically is when the user logs out.
Currently there are no ways to receive this event.

### Solutions

- Add a `observeLogout(): Flow<LogoutReason>` function to `LogoutRepository`.
- Add a `onLogout(reason: LogoutReason)` to `LogoutRepository`.

The idea is that `onLogout` should be called by entities (like the UseCases) with a reason. This event can be observed and handled by the SyncManager later.

#### Test Coverage

- [X] I have added automated test to this contribution

----
#### PR Post Merge Checklist for internal contributors

 - [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by mchenani** [comment]
Generally it's simple and clean, but actually we need more that that! Like we need to store the reason [ what I did in my expiration manager] and more than that like soft / hard logout and ... 
I think if we can wait till finalise the ticket I have we can achieve more and the full result, but if you're blocked and need that we can merge it now and then I apple my changes to that.
In the end I'll add an observer like that but with different backgroun.

**Comment by vitorhugods** [comment]
> Generally it's simple and clean, but actually we need more that that! Like we need to store the reason [ what I did in my expiration manager] and more than that like soft / hard logout and ... 
> I think if we can wait till finalise the ticket I have we can achieve more and the full result, but if you're blocked and need that we can merge it now and then I apple my changes to that.
> In the end I'll add an observer like that but from different background.
> 

I am a _bit_ blocked by this.
I hadn't thought of persisting it and indeed that's pretty much needed, but I guess it's simple to add it later, no?
We can add a table/DAO or SharedPrefs and instead of just propagating the event in memory with this Channel, we can observe the storage and _voilà_.

What do you think?

**Review by mchenani — APPROVED** [review]
As we discussed, that would be temporary to unblock you till I got my PR merged! LGTM atm 🚀

**Pull request merged**
